### PR TITLE
refactor: navbar's social icons items

### LINKED ISSUE #1448: [Refactor]: header social items
### File Name

https://github.com/reactplay/react-play/blob/main/src/common/header/HeaderNav.jsx#L127

### Reason for Refactoring the Code

we need to refactor the navbar's social icons links using `.map` method 

```
      <li>
            <Link
              className="app-header-btn app-header-btn--default"
              data-testid="leaderboard-btn"
              title="Leader Board"
              to="/leaderboard"
            >
              <BsTrophyFill className="icon idea-icon" />
              <span className="btn-label">Leader Board</span>
            </Link>
          </li>
          <li>
            <Link
              className="app-header-btn app-header-btn--default"
              data-testid="ideas-btn"
              title="Play Ideas"
              to="/ideas"
            >
              <FaLightbulb className="icon idea-icon" />
              <span className="btn-label">Idea</span>
            </Link>
          </li>
          <li>
            <a
              className="app-header-btn app-header-btn--default"
              data-testid="github-btn"
              data-umami-event="github-button"
              href="https://github.com/reactplay/react-play"
              rel="noopener noreferrer"
              target="_blank"
              title="GitHub page"
            >
              <BsGithub className="icon github-icon" />
              <span className="btn-label">GitHub</span>
            </a>
          </li>
          <li>
            <a
              className="app-header-btn app-header-btn--default"
              data-testid="twitter-btn"
              href="https://twitter.com/reactplayio"
              rel="noopener noreferrer"
              target="_blank"
              title="Twitter Page"
            >
              <FaXTwitter className="icon twitter-icon" />
              <span className="btn-label">Twitter</span>
            </a>
          </li>
          <li>
            <button
              className="app-header-btn app-header-btn--default"
              data-testid="share-btn"
              title="Show love"
              onClick={handleClick}
            >
              <IoHeartSharp className="icon share-icon" />
              <span className="btn-label">Share</span>
            </button>
          </li>
```

### Record

- [X] I have checked the existing [issues](https://github.com/reactplay/react-play/issues)
- [X] I have read the [Contributing Guidelines](https://github.com/reactplay/react-play/blob/main/CONTRIBUTING.md)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/reactplay/react-play/blob/main/CODE_OF_CONDUCT.md)
- [ ] I want to work on this issue

### PR DESCRIPTION
**Pull request by Shivam-Katare**
# Description

Refactor the navbar's social icons code. Removed the repetition part, and instead of that used the mapping to display all icons.

Fixes #1448 

## Type of change

- Code Refactoring

# How Has This Been Tested?

- Tested this change over Chrome Browser and Microsoft Edge

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

# Screenshots or example output

image

### CODE REVIEW

**Comment by Shivam-Katare** [comment]
@priyankarpal, let me know if any changes are required in it.

File: src/common/header/HeaderNav.jsx [hunk 1]
```diff
@@ -30,6 +30,55 @@ const HeaderNav = ({ showBrowse }) => {
 
   const modalClose = () => setShowShareModal(!showShareModal);
 
+  const links = [
```
Comment (priyankarpal) [inline src/common/header/HeaderNav.jsx:33]: rename to `NavLinks`
```suggestion
  const NavLinks = [
```

File: src/common/header/HeaderNav.jsx [hunk 2]
```diff
@@ -124,66 +173,28 @@ const HeaderNav = ({ showBrowse }) => {
...
-              className="app-header-btn app-header-btn--default"
-              data-testid="share-btn"
-              title="Show love"
-              onClick={handleClick}
-            >
-              <IoHeartSharp className="icon share-icon" />
-              <span className="btn-label">Share</span>
-            </button>
-          </li>
+          {links.map((link, index) => {
```
Comment (priyankarpal) [inline src/common/header/HeaderNav.jsx:176]:
```suggestion
          {NavLinks.map((link, index) => {
```

File: src/common/header/HeaderNav.jsx [hunk 3]
```diff
@@ -124,66 +173,28 @@ const HeaderNav = ({ showBrowse }) => {
...
+            const Component = link.type === 'Link' ? Link : link.type;
```
Comment (priyankarpal) [inline src/common/header/HeaderNav.jsx:177]:
```suggestion
            const Component = NavLink.type === 'Link' ? Link : NavLink.type;
```

**Review by priyankarpal — CHANGES_REQUESTED** [review]
rename all `link` to `NavLink` @Shivam-Katare

File: src/common/header/HeaderNav.jsx (hunk 3 above)
Comment (Shivam-Katare) [reply src/common/header/HeaderNav.jsx:177]: Sure.

**Comment by Shivam-Katare** [comment]
@priyankarpal, I renamed all `links` to `NavLinks` and also, Rename the `icon` to `Icon` in NavLink mapping to stick to the **PascalCase convention** for JSX components

**Comment by priyankarpal** [comment]
> @priyankarpal, I renamed all `links` to `NavLinks` and also, Rename the `icon` to `Icon` in NavLink mapping to stick to the **PascalCase convention** for JSX components

thank you

**Review by priyankarpal — APPROVED** [review]
Looks Good 🎉

**Pull request merged**